pple-Mail-1--66779474
Content-Transfer-Encoding: 7bit
Content-Type: text/plain;
	charset=US-ASCII;
	delsp=yes;
	format=flowed


On 20 Apr 2009, at 12:12, Axel Thimm wrote:

> Hi,
>
> current Fedora rawhide (to become F11) has the following components:
>
> glibc-2.9.90-19
> gcc-4.4.0-0.34
> kernel-2.6.29.1-100.fc11
>
> The build (1.4.10 and 1.4.8) fails with
>
> gcc -pipe  -O2 -I/builddir/openafs-1.4.10/src/config -I. -I. -I/ 
> builddir/openafs-1.4.10/include -I/builddir/openafs-1.4.10/include/ 
> afs -I/builddir/openafs-1.4.10/include/rx -I/builddir/ 
> openafs-1.4.10 -I/builddir/openafs-1.4.10/src -I/builddir/ 
> openafs-1.4.10/src -g -O2 -D_LARGEFILE64_SOURCE  -c admin_tools.c
> In file included from admin_tools.c:30:
> /usr/include/stdio.h:399:66: error: macro "dprintf" passed 3  
> arguments, but takes just 2
> admin_tools.c: In function 'MyBeforeProc':
> admin_tools.c:1470: warning: passing argument 2 of 'ktc_GetToken'  
> from incompatible pointer type
> /builddir/openafs-1.4.10/include/afs/auth.h:36: note: expected  
> 'struct ktc_token *' but argument is of type 'struct ktc_principal *'

Could you try the attached (untested) patch...

In future, Bcc-ing openafs-bugs (I've just done so) with these kinds  
of reports helps us track them, and their patches, better.

Thanks,

Simon.

--Apple-Mail-1--66779474
Content-Transfer-Encoding: 7bit
Content-Type: application/octet-stream;
	x-unix-mode=0644;
	name=afs_dprintf
Content-Disposition: attachment;
	filename=afs_dprintf

Index: stable/openafs/src/bucoord/bc.p.h
===================================================================
--- stable.orig/openafs/src/bucoord/bc.p.h	2009-03-23 12:39:24.000000000 +0000
+++ stable/openafs/src/bucoord/bc.p.h	2009-04-20 13:55:19.000000000 +0100
@@ -144,4 +144,4 @@
 #define	BC_MAXSIMDUMPS	    64
 #define BC_MAXPORTS	    128	/* max number of port offsets for volrestore */
 /* debugging support */
-#define	dprintf(x)
+#define	afs_dprintf(x)
Index: stable/openafs/src/bucoord/ubik_db_if.c
===================================================================
--- stable.orig/openafs/src/bucoord/ubik_db_if.c	2009-03-23 12:49:14.000000000 +0000
+++ stable/openafs/src/bucoord/ubik_db_if.c	2009-04-20 13:55:43.000000000 +0100
@@ -601,7 +601,7 @@
 
     fileSize = (afs_int32) filesize(ctPtr->textStream);
 
-    dprintf(("filesize is %d\n", fileSize));
+    afs_dprintf(("filesize is %d\n", fileSize));
 
     rewind(ctPtr->textStream);
 
@@ -1390,7 +1390,7 @@
 	ERROR(errno);
 #endif
 
-    dprintf(("file is %s\n", tmpFileName));
+    afs_dprintf(("file is %s\n", tmpFileName));
 
   normal_exit:
     return code;
Index: stable/openafs/src/config/debug.h
===================================================================
--- stable.orig/openafs/src/config/debug.h	2003-07-16 00:14:55.000000000 +0100
+++ stable/openafs/src/config/debug.h	2009-04-20 13:56:31.000000000 +0100
@@ -112,21 +112,21 @@
 
 #if defined(AFS_DEBUG)
 #if defined(lint)
-#define dprintf(flag, str) printf str
+#define afs_dprintf(flag, str) printf str
 #define dlprintf(flag, level, str) printf str
 #define dmprintf(flag, bit, str) printf str
 #else /* lint */
-#define dprintf(flag, str) \
+#define afs_dprintf(flag, str) \
       (void)((flag) ? \
 	     ( osi_dp str, osi_dp("\t%s, %d\n", __FILE__, __LINE__)):0)
-#define dlprintf(flag, level, str) dprintf(((flag) >= (level)), str)
-#define dmprintf(flag, bit, str) dprintf(((flag)&(1<<((bit)-1))), str)
+#define dlprintf(flag, level, str) afs_dprintf(((flag) >= (level)), str)
+#define dmprintf(flag, bit, str) afs_dprintf(((flag)&(1<<((bit)-1))), str)
 
 #endif /* lint */
 
 #else /* AFS_DEBUG */
 
-#define dprintf(flag, str)
+#define afs_dprintf(flag, str)
 #define dlprintf(flag, level,str)
 #define dmprintf(flag, bit, str)
 
Index: stable/openafs/src/sgistuff/rcmd.c
===================================================================
--- stable.orig/openafs/src/sgistuff/rcmd.c	2005-08-16 19:00:44.000000000 +0100
+++ stable/openafs/src/sgistuff/rcmd.c	2009-04-20 13:57:54.000000000 +0100
@@ -55,8 +55,8 @@
 #endif /* defined(AFS_HPUX_ENV) */
 #ifdef	TCP_DEBUG
 #include <sys/syslog.h>
-#	define	DPRINTF(args)	dprintf args
-dprintf(args)
+#	define	DPRINTF(args)	afs_dprintf args
+afs_dprintf(args)
      char *args;
 {
     char **argv;

--Apple-Mail-1--66779474--

